dissect/target/filesystems/zip.py

```diff
@@ -161,6 +161,11 @@ def lstat(self) -> fsutil.stat_result:
         elif not self.entry.is_dir() and not stat.S_ISREG(mode):
             mode = stat.S_IFREG | mode
 
+        if self.entry.date_time[0] < 1980 or self.entry.date_time[:3] == (1980, 0, 0):
+            self.entry.date_time = (1980, 1, 1, 0, 0, 0)
+        elif self.entry.date_time[0] > 2107:
+            self.entry.date_time = (2107, 12, 31, 23, 59, 59)
```

Comment:
```suggestion
        date_time = self.entry.date_time
        if date_time[0] < 1980 or date_time[:3] == (1980, 0, 0):
            date_time = (1980, 1, 1, 0, 0, 0)
        elif date_time[0] > 2107:
            date_time = (2107, 12, 31, 23, 59, 59)
```

And then change the line below (I can't put a suggestion there) to this:
```python
datetime(*date_time, tzinfo=timezone.utc).timestamp(),
```